src/ott/math/matrix_square_root.py

```diff
@@ -235,18 +235,37 @@ def sqrtm_bwd(
 # These functions have lower complexity gradients than sqrtm.
 
 
-@jax.custom_vjp
-def sqrtm_only(x: jnp.ndarray) -> jnp.ndarray:
-  return sqrtm(x)[0]
-
-
-def sqrtm_only_fwd(x: jnp.ndarray) -> Tuple[jnp.ndarray, jnp.ndarray]:
-  sqrt_x = sqrtm(x)[0]
+@functools.partial(jax.custom_vjp, nondiff_argnums=(1, 2, 3, 4, 5))
+def sqrtm_only(
+    x: jnp.ndarray,
+    threshold: float = 1e-6,
+    min_iterations: int = 0,
+    inner_iterations: int = 10,
+    max_iterations: int = 1000,
+    regularization: float = 1e-6
+) -> jnp.ndarray:
+  return sqrtm(
+      x, threshold, min_iterations, inner_iterations, max_iterations,
+      regularization
+  )[0]
+
+
+def sqrtm_only_fwd(
+    x: jnp.ndarray, threshold: float, min_iterations: int,
+    inner_iterations: int, max_iterations: int, regularization: float
+) -> Tuple[jnp.ndarray, jnp.ndarray]:
+  sqrt_x = sqrtm(
+      x, threshold, min_iterations, inner_iterations, max_iterations,
+      regularization
+  )[0]
   return sqrt_x, sqrt_x
 
 
-def sqrtm_only_bwd(sqrt_x: jnp.ndarray,
-                   cotangent: jnp.ndarray) -> Tuple[jnp.ndarray]:
+def sqrtm_only_bwd(
+    threshold: float, min_iterations: int, inner_iterations: int,
+    max_iterations: int, regularization: float, sqrt_x: jnp.ndarray,
+    cotangent: jnp.ndarray
+) -> Tuple[jnp.ndarray]:
```

Comment:
Can you please delete the unused arguments?

Reply:
yes!